src/dev/graph/layout.rs

```diff
@@ -83,28 +84,36 @@ impl CircuitLayout {
     /// Renders the given circuit on the given drawing area.
     pub fn render<F: Field, ConcreteCircuit: Circuit<F>, DB: DrawingBackend>(
         self,
+        k: usize,
         circuit: &ConcreteCircuit,
         drawing_area: &DrawingArea<DB, Shift>,
     ) -> Result<(), DrawingAreaErrorKind<DB::ErrorType>> {
         use plotters::coord::types::RangedCoordusize;
         use plotters::prelude::*;
 
+        let n = 1 << k;
         // Collect the layout details.
         let mut cs = ConstraintSystem::default();
         let config = ConcreteCircuit::configure(&mut cs);
-        let mut layout = Layout::default();
+        let mut layout = Layout::new(n, cs.num_selectors);
         ConcreteCircuit::FloorPlanner::synthesize(
             &mut layout,
             circuit,
             config,
             cs.constants.clone(),
         )
         .unwrap();
+        let (cs, selector_polys) = cs.compress_selectors(layout.selectors);
+        let non_selector_fixed_columns = cs.num_fixed_columns - selector_polys.len();
```

Comment:
The selector columns of the layout diagram are actually rather broken after this change (due to how regions end up overlapping), but I'll fix that in a separate PR.